pkg/function/kubeops.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"context"
+	"encoding/json"
+
+	kanister "github.com/kanisterio/kanister/pkg"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/param"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	_ = kanister.Register(&kubeops{})
+}
+
+var (
+	_ kanister.Func = (*kubeops)(nil)
+)
+
+const (
+	// KubeOpsFuncName gives the name of the function
+	KubeOpsFuncName = "KubeOps"
+	// KubeOpsSpecsArg provides resource specs yaml
+	KubeOpsSpecsArg = "specs"
```

Comment:
Should this be singular since we support a single `spec` string?
If yes, let's make it `spec` because we cannot change it once it's used in blueprints

Reply:
yes, that makes sense